### PR TITLE
Swift 5.1 images

### PR DESCRIPTION
**Pull request by gwynne**
Built on @ianpartridge's original 5.1 effort. Adds signature verification, ability to override the key fingerprint used for verification, and ability to override the base URL used for downloads.

### CODE REVIEW

**Comment by gwynne** [comment]
@swift-ci test

**Comment by gwynne** [comment]
@swift-ci test

**Comment by gwynne** [comment]
@swift-ci please test

**Comment by gwynne** [comment]
Test failures were for the ever-wonderful failure to bind requested address for the keyservers. The images build correctly locally.

**Comment by gwynne** [comment]
cc @shahmishal

**Comment by ianpartridge** [comment]
Thanks! For the slim images we only want the runtime libraries, not the whole toolchain. See https://github.com/apple/swift-docker/blob/master/5.0/ubuntu/16.04/slim/Dockerfile#L40

**Comment by gwynne** [comment]
@ianpartridge Yikes, thanks! Nothing like a little copypasta to mess you up 😅 I also fixed the caching of the images around `curl` and `gnupg` (with the latter turning out to only be needed by the slim 18.04 image, since Xenial has it as a dependency of `apt` and in non-slim Bionic we get it via `dpkg-dev`'s recommends via our `pkg-config`)

**Comment by gwynne** [comment]
@swift-ci please test

**Comment by gwynne** [comment]
@swift-ci please test

**Review by ianpartridge — APPROVED** [review]
Looks good to me! Would welcome extra pairs of eyes cc: @hamin @shahmishal

**Comment by proggeramlug** [comment]
No rush, but I would love to have this image available soon. :D

**Comment by tonyarnold** [comment]
I could also really use this image when it's convenient to release it ❤️

File: 5.1/ubuntu/16.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+# Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
+
+# pub   rsa4096 2017-11-07 [SC] [expires: 2019-11-07]
+#       8513 444E 2DA3 6B7C 1659  AF4D 7638 F1FB 2B2B 08C4
+# uid           [ unknown] Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>
+ARG SWIFT_SIGNING_KEY=8513444E2DA36B7C1659AF4D7638F1FB2B2B08C4
+ARG SWIFT_PLATFORM=ubuntu16.04
+ARG SWIFT_BRANCH=swift-5.1-branch
+ARG SWIFT_VERSION=swift-5.1-DEVELOPMENT-SNAPSHOT-2019-08-24-a
```
Comment (shahmishal) [inline 5.1/ubuntu/16.04/Dockerfile:30]:
```suggestion
ARG SWIFT_VERSION=swift-5.1-RELEASE
```

File: 5.1/ubuntu/16.04/slim/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
+
+# pub   rsa4096 2017-11-07 [SC] [expires: 2019-11-07]
+#       8513 444E 2DA3 6B7C 1659  AF4D 7638 F1FB 2B2B 08C4
+# uid           [ unknown] Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>
+ARG SWIFT_SIGNING_KEY=8513444E2DA36B7C1659AF4D7638F1FB2B2B08C4
+ARG SWIFT_PLATFORM=ubuntu16.04
+ARG SWIFT_BRANCH=swift-5.1-branch
+ARG SWIFT_VERSION=swift-5.1-DEVELOPMENT-SNAPSHOT-2019-08-24-a
```
Comment (shahmishal) [inline 5.1/ubuntu/16.04/slim/Dockerfile:21]:
```suggestion
ARG SWIFT_VERSION=swift-5.1-RELEASE
```

File: 5.1/ubuntu/18.04/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
+
+# pub   rsa4096 2017-11-07 [SC] [expires: 2019-11-07]
+#       8513 444E 2DA3 6B7C 1659  AF4D 7638 F1FB 2B2B 08C4
+# uid           [ unknown] Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>
+ARG SWIFT_SIGNING_KEY=8513444E2DA36B7C1659AF4D7638F1FB2B2B08C4
+ARG SWIFT_PLATFORM=ubuntu18.04
+ARG SWIFT_BRANCH=swift-5.1-branch
+ARG SWIFT_VERSION=swift-5.1-DEVELOPMENT-SNAPSHOT-2019-08-24-a
```
Comment (shahmishal) [inline 5.1/ubuntu/18.04/Dockerfile:30]:
```suggestion
ARG SWIFT_VERSION=swift-5.1-RELEASE
```

File: 5.1/ubuntu/18.04/slim/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+# Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
+
+# pub   rsa4096 2017-11-07 [SC] [expires: 2019-11-07]
+#       8513 444E 2DA3 6B7C 1659  AF4D 7638 F1FB 2B2B 08C4
+# uid           [ unknown] Swift Automatic Signing Key #2 <swift-infrastructure@swift.org>
+ARG SWIFT_SIGNING_KEY=8513444E2DA36B7C1659AF4D7638F1FB2B2B08C4
+ARG SWIFT_PLATFORM=ubuntu18.04
+ARG SWIFT_BRANCH=swift-5.1-branch
+ARG SWIFT_VERSION=swift-5.1-DEVELOPMENT-SNAPSHOT-2019-08-24-a
```
Comment (shahmishal) [inline 5.1/ubuntu/18.04/slim/Dockerfile:21]:
```suggestion
ARG SWIFT_VERSION=swift-5.1-RELEASE
```

**Comment by ianpartridge** [comment]
@shahmishal your changes look good!

**Comment by shahmishal** [comment]
Created new PR with Swift 5.1 release packages - https://github.com/apple/swift-docker/pull/147

**Pull request merged**